### PR TITLE
[5.09.30] Bricked Blast Furnace

### PR DESCRIPTION
**Pull request by JohannesGaessler**
As a way of rebalancing the Bronze Age I have added a Blast Furnace that is built from bricks.
This eliminates the need to gather large amounts of Bronze for a Bronze Blast Furnace.

The Bricked Blast Furnace has the same structure as the Bronze Blast Furnace and is built from Firebricks.
Firebricks are made from clay, but they require more complex processing than vanilla Bricks.

To build a Bricked Blast Furnace the player needs to build a Steam Compressor and a Steam Hammer/Steam Macerator.
The first Bricked Blast Furnace requires a total of 130 Clay blocks to build.
This is comparable to the 150 Bronze required for a Bronze Blast Furnace, but clay is found much more common than Copper and Tin, therefore reducing the amount of time the player needs to spend on searching for a vein.

On another note, NEI recipes for the primitive blast furnaces have been added.

### CODE REVIEW

**Comment by Rongmario** [comment]
There is a complete GT version up for compile with this and the OilRefinery update on my fork of GT5U.

File: src/main/java/gregtech/common/GT_RecipeAdder.java [hunk 1]
```diff
@@ -143,6 +145,26 @@ public boolean addBlastRecipe(ItemStack aInput1, ItemStack aInput2, FluidStack a
...
+        }
+        if (aCoalAmount <= 0) {
+        	return false;
+        }
+        if ((aDuration = GregTech_API.sRecipeFile.get("primitiveblastfurnace", aInput1, aDuration)) <= 0) {
+            return false;
+        }
+        ItemStack[] coalStacks = new ItemStack[]{Materials.Coal.getGems(aCoalAmount), Materials.Coal.getDust(aCoalAmount), Materials.Charcoal.getGems(aCoalAmount), Materials.Charcoal.getDust(aCoalAmount)};
+        for (ItemStack coalStack : coalStacks) {
+        	GT_Recipe.GT_Recipe_Map.sPrimitiveBlastRecipes.addRecipe(true, new ItemStack[]{aInput1, aInput2, coalStack}, new ItemStack[]{aOutput1, aOutput2, Materials.Ash.getDustTiny(aCoalAmount)}, null, null, null, null, aDuration, 0, 0);
```
Comment (draknyte1) [inline src/main/java/gregtech/common/GT_RecipeAdder.java]: Normal Coal and Coke produce different types of ash..

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_PrimitiveBlastFurnace.java [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+		this.mProgresstime += aProgress;
+		return this.mMaxProgresstime - this.mProgresstime;
+	}
+
+	public boolean allowCoverOnSide(byte aSide, GT_ItemStack aCoverID) {
+		return (GregTech_API.getCoverBehavior(aCoverID.toStack()).isSimpleCover()) && (super.allowCoverOnSide(aSide, aCoverID));
+	}
+
+	public MetaTileEntity newMetaEntity(IGregTechTileEntity aTileEntity) {
+		return new GT_MetaTileEntity_BronzeBlastFurnace(this.mName);
```
Comment (draknyte1) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_PrimitiveBlastFurnace.java]: Is this intended?

Comment (Techlone) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_PrimitiveBlastFurnace.java]: The copy-paste mistake
Happens

**Review by draknyte1 — CHANGES_REQUESTED** [review]
So, 95% of it looks good.
Just be sure to change the things noted and add a config and I see no issues otherwise.

Also be sure the BBF uses the BBF GUI and TE, whilst the bricked one uses the Bricked GUI and TE.
Don't remember which the BBF uses, the diffing is ugh online due to everything having moved a line.

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_PrimitiveBlastFurnace.java (hunk 2 above)
Comment (draknyte1) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_PrimitiveBlastFurnace.java]: Just pointing things out during my review.

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_BrickedBlastFurnace.java [hunk 3]
```diff
@@ -29,7 +29,7 @@ public MetaTileEntity newMetaEntity(IGregTechTileEntity aTileEntity) {
                 "Controller Block for the Bricked Blast Furnace",
                 "Useable for Steel and general Pyrometallurgy",
                 "Size(WxHxD): 3x4x3 (Hollow, with opening on top)",
-                "Built from 32 Bricks",
+                "Built from 32 Firerick blocks",
```
Comment (Techlone) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_BrickedBlastFurnace.java]: *fire**b**rick

**Pull request merged**